pysb/importers/bngl.py

```diff
@@ -0,0 +1,319 @@
+                cplx_pats.append(ComplexPattern(self._parse_species(mp),
+                                                compartment=None,
+                                                match_once=match_once))
+            self.observable(o_name,
+                            ReactionPattern(cplx_pats),
+                            match=o.get('type').lower())
+
+    def parse_initials(self):
+        for i in self.x.iterfind(_ns('{0}ListOfSpecies/{0}Species')):
+            if i.get('Fixed') is not None and i.get('Fixed') == "1":
+                warnings.warn('Species %s is fixed, but will be treated as '
+                              'an ordinary species in PySB.' % i.get('name'))
+
+            value_param = i.get('concentration')
+            try:
+                value = float(value_param)
+                # Need to create a new parameter for the initial conc. literal
+                name = re.sub('[^\w]+', '_', i.get('name').replace(
+                    ')', '').replace('(', '')) + '_0'
+                try:
+                    value_param = self.parameter(name, value)
+                except ValueError as ve:
+                    raise BnglImportError(str(ve))
+            except ValueError:
+                # Retrieve existing parameter or (constant) expression
+                try:
+                    value_param = self.model.parameters[i.get('concentration')]
+                except KeyError:
+                    value_param = self.model.expressions[i.get(
+                        'concentration')]
+            mon_pats = self._parse_species(i)
+            species_cpt = self.model.compartments.get(i.get('compartment'))
+            self.initial(ComplexPattern(mon_pats, species_cpt), value_param)
+
+    def parse_compartments(self):
+        for c in self.x.iterfind(_ns('{0}ListOfCompartments/{0}compartment')):
+            cpt_size = None
+            if c.get('size'):
+                cpt_size = self.parameter('%s_size' % c.get('id'),
+                                          c.get('size'))
+            cpt_parent = None
+            if c.get('outside'):
+                cpt_parent = self.model.compartments[c.get('outside')]
+            self.compartment(name=c.get('id'),
+                             parent=cpt_parent,
+                             dimension=int(c.get('spatialDimensions')),
+                             size=cpt_size)
+
+    def _parse_rate_law(self, rl):
+        if rl.get('type') == 'Ele':
+            rate_law_name = rl.find(_ns('{0}ListOfRateConstants/'
+                                        '{0}RateConstant')).get('value')
+            try:
+                return self.model.parameters[rate_law_name]
+            except KeyError:
+                return self.model.expressions[rate_law_name]
+        elif rl.get('type') == 'Function':
+            return self.model.expressions[rl.get('name')]
+        else:
+            raise BnglImportError('Rate law %s has unknown type %s' %
```

Comment:
Shouldn't we be able to implement `MM` and `Sat` with Expressions? It's fine to punt on it for now but it seems like we could handle these if we wanted to.